src/app/generate_keypair/generate_keypair.ml

```diff
@@ -0,0 +1,35 @@
+open Core
+open Async
+open Signature_lib
+
+let main privkey_path =
+  let%map {public_key; _} =
+    Secrets.Keypair.Terminal_stdin.read_exn privkey_path
+  in
+  printf "%s\n%!"
+    (Public_key.Compressed.to_base58_check (Public_key.compress public_key))
+
+let handle_exception_nicely (type a) (f : unit -> a Deferred.t) () :
+    a Deferred.t =
+  match%bind Deferred.Or_error.try_with ~extract_exn:true f with
+  | Ok e ->
+      return e
+  | Error e ->
+      eprintf "Error: %s" (Error.to_string_hum e) ;
+      exit 4
+
+let cmd =
+  Command.async ~summary:"Generate a new public-key/private-key pair"
+    (let open Command.Let_syntax in
+    let%map_open privkey_path = Cli_lib.Flag.privkey_write_path in
+    handle_exception_nicely
```

Comment:
Style nit: to help make the code a little more readable, I would recommend lifting the entire `@@ fun () ->` portion of this expression up into a definition such as `let main () = ...`. Then this can read nicer as just `handle_exception_nicely main`. It seems like you may have started doing that with your `let main privkey_path = ...` definition above, but it's currently unused.

Reply:
right, the existing `main` isn't being used, it can be deleted.